src/tools/grpc_server.cc

```diff
@@ -174,14 +175,10 @@ class add_match {
     void operator()(const match_result *m) const {
         // Avoid a duplicate if a line is returned once from the
         // tags search then again during the main corpus search.
-        std::string key = std::string(m->file->tree->name)
-            + " " + std::string(m->file->tree->version)
-            + " " + std::string(m->file->path)
-            + " " + std::to_string(m->lno);
-        bool was_inserted = unique_lines_->insert(key).second;
-        if (!was_inserted) {
+        line_set_key k(m->file, m->lno);
```

Comment:
You can just inline this into `unique_lines_->insert(make_pair(m->file, m->lno))` and skip the declaration.

Reply:
Neat! How does the destructor wind up getting called on the pair in that case, though? I'd created the declaration so that I could build it on the stack and thus know when it got deallocated.

Reply:
Temporaries are constructed on the stack and destructed at the end of the containing statement, so just after the call to `insert`. It's a fairly common C++ idiom to use that fact in situations like this.